public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import { InfluxQuery, InfluxQueryTag } from '../../types';
+import { getTemplateSrv } from '@grafana/runtime';
+import InfluxDatasource from '../../datasource';
+import { FromSection } from './FromSection';
+import { TagsSection } from './TagsSection';
+import { PartListSection } from './PartListSection';
+import { OrderByTimeSection } from './OrderByTimeSection';
+import { InputSection } from './InputSection';
+import { unwrap } from './unwrap';
+import {
+  getAllMeasurements,
+  getAllPolicies,
+  getFieldKeysForMeasurement,
+  getTagKeysForMeasurementAndTags,
+  getTagValues,
+} from '../../influxQLMetadataQuery';
+import {
+  normalizeQuery,
+  addNewSelectPart,
+  removeSelectPart,
+  addNewGroupByPart,
+  removeGroupByPart,
+} from '../queryUtils';
+import { FormatAsSection } from './FormatAsSection';
+import { SectionLabel } from './SectionLabel';
+import { SectionFill } from './SectionFill';
+import { DEFAULT_RESULT_FORMAT } from '../constants';
+import { getNewSelectPartOptions, getNewGroupByPartOptions, makePartList } from './partListUtils';
+
+type Props = {
+  query: InfluxQuery;
+  onChange: (query: InfluxQuery) => void;
+  onRunQuery: () => void;
+  datasource: InfluxDatasource;
+};
+
+function getTemplateVariableOptions() {
+  return (
+    getTemplateSrv()
+      .getVariables()
+      // we make them regex-params, i'm not 100% sure why.
```

Comment:
I think this is to indicate that the value will be interpolated using [Regex formating](https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/#regex). The only thing that is missing is to switch the operator to `=~` when a variable is selected to use it correctly in the query:

| Angular | React |
|---------|-------|
|![template-2](https://user-images.githubusercontent.com/745532/116418386-615e3480-a83c-11eb-92d7-d20477dc5e76.gif) | ![template-1](https://user-images.githubusercontent.com/745532/116418450-71761400-a83c-11eb-8af5-c11e3039ffc9.gif) |

Reply:
yes, i decided to not do that because it feels a little too magic to me. also, for example, if you have `cpu != cpu1`, and now choose instead of `cpu` `/^cpu.*$/`, we are switching from `!=` to `~=` (from not-string-equal to yes-regex-equal), which i'm not sure if it's better. but i can change this if you think it's better that way.

Reply:
Good point, I didn't notice that switch from `!=` to `=~` :D

When a variable is used as a value then only `=~` and `!~` operators make sense. In Angular version the list is actually limited to those two operators. I think it's quite useful, it's easy to forget to update the operator. But as you mentioned, at the same time it may not be the operator that the user wants to use. I'm not sure what's better in this case - (1) change things automagically to something valid (like in Angular), or (2) give the user full freedom and selecting anything. 

I think less magic is better, it's always hard to guess what the user really wants. Would it be difficult to somehow let the user know that the selected operator doesn't make sense in this case? For example change the colour of `=` to red? If it's too complex I think you can leave it as it is now but I think it'd be worth improving it somehow later.

Reply:
i looked into it some more, and could not find a nice solution there, and i know i was againts it first, but now i think the pragmatic choice is just to do what angular did (so the new solution is at least not worse than the old), so i implemented the automatic-operator-switching 😁 i still don't like it and we should improve it, but i could not find a simple way to improve the situation.
i hope this is acceptable for you, if not, feel free to reopen this conversation and we can find another way.